mne/preprocessing/ica.py

```diff
@@ -2079,3 +2079,239 @@ def _band_pass_filter(ica, sources, target, l_freq, h_freq, verbose=None):
         raise ValueError('Must specify both pass bands')
 
     return sources, target
+
+
+@verbose
+def corrmap(icas, template, threshold="auto", name="bads",
+            plot=True, inplace=False, ch_type="eeg"):
+    """Corrmap (Viola et al. 2009 Clin Neurophysiol) identifies the best group
+    match to a supplied template. Typically, feed it a list of fitted ICAs and
+    a template IC, for example, the blink for the first subject, to identify
+    specific ICs across subjects.
+
+    The specific procedure consists of two iterations. In a first step, the
+    maps best correlating with the template are identified. In the step, the
+    analysis is repeated with the mean of the maps identified in the first
+    stage.
+
+    Outputs a list of fitted ICAs with the indices of the marked ICs in a
+    specified field.
+
+    The original Corrmap website: www.debener.de/corrmap/corrmapplugin1.html
+
+    Parameters
+    ----------
+    icas : list
+        A list of fitted ICAs.
+    template : (int, int) tuple
+        Index of the list member from which the template should be obtained,
+        and the index of the IC.
+    threshold : "auto" | list of floats > 0 < 1 | float > 0 < 1 | float > 1
+        Correlation threshold for identifying ICs
+        If "auto": search for the best map by trying all correlations between
+        0.6 and 0.95. In the original proposal, lower values are considered,
+        but this is not yet implemented.
+        If list of floats: search for the best map in the specified range of
+        correlation strengths.
+        If float > 0: select ICs correlating better than this.
+        If float > 1: use find_outliers to identify ICs within subjects (not in
+        original Corrmap)
+        Defaults to "auto".
+    name : str
+        Categorised ICs are stored in a default dictionary "labels". This
+        parameter gives the key under which found ICs will be stored.
+        Defaults to "bads".
+    plot : bool
+        Should constructed template and selected maps be plotted?
+    ch_type : 'mag' | 'grad' | 'planar1' | 'planar2' | 'eeg'
+            The channel type to plot. Defaults to 'eeg'.
+
+    Returns
+    -------
+    new_icas  :  list
+        Returns a list of fitted ICs, enrichened with the indices of the
+        selected maps in the field specified by the name keyword.
+    """
+
+    def vcorrcoef(X, y):  # vectorised correlation, by Jon Herman
+        Xm = np.reshape(np.mean(X, axis=1), (X.shape[0], 1))
+        ym = np.mean(y)
+        r_num = np.sum((X - Xm) * (y - ym), axis=1)
+        r_den = np.sqrt(np.sum((X - Xm) ** 2, axis=1) * np.sum((y - ym) ** 2))
+        r = r_num / r_den
+        return r
+
+    def get_ica_map(ica, components=None):
+        if components is None:
+            components = range(ica.n_components_)
+        maps = np.dot(ica.mixing_matrix_[:, components].T,
+                      ica.pca_components_[:ica.n_components_])
```

Comment:
did you check this is correct? The sphering / prewhitening step is not considered here. compare with plot topo viz code